go.work

```diff
@@ -1,6 +0,0 @@
-go 1.20
```

Comment:
@tomhjp FYI

According to https://go.googlesource.com/proposal/+/master/design/45713-workspace.md
> These go.work files should not be checked into the repositories so that they don‘t override the workspaces users explicitly define. Checking in go.work files could also lead to CI/CD systems not testing the actual set of version requirements on a module and that version requirements among the repository’s modules are properly incremented to use changes in the modules.

Safe to remove? It caused some weirdness in our automated dependency update workflow. See PR here: https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/205

Reply:
Oops, TIL. Thanks for the link! Yeah it's safe to remove, I only added it as a convenience to make `gopls` able to understand multiple modules.